common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -355,52 +378,92 @@ protected boolean removeTask(Runnable task) {
      * @return {@code true} if and only if at least one task was run
      */
     protected boolean runAllTasks() {
+        assert inEventLoop();
         boolean fetchedAll;
+        boolean ranAtLeastOne = false;
+
         do {
             fetchedAll = fetchFromScheduledTaskQueue();
-            Runnable task = pollTask();
-            if (task == null) {
-                return false;
+            if (runAllTasksFrom(taskQueue)) {
+                ranAtLeastOne = true;
             }
+        } while (!fetchedAll); // keep on processing until we fetched all scheduled tasks.
 
-            for (;;) {
-                try {
-                    task.run();
-                } catch (Throwable t) {
-                    logger.warn("A task raised an exception.", t);
-                }
+        if (ranAtLeastOne) {
```

Comment:
is this necessary any more? it seems like `setLastExecutionTime` will get called on the last invocation of `runAllTasksFrom(...)` ... maybe we can just get rid of this boolean and simplify the loop above a bit?